### PR TITLE
Home end point

### PR DESCRIPTION
**Pull request by YousefQawasmeh**

### CODE REVIEW

File: config.env [hunk 1]
```diff
@@ -0,0 +1 @@
+DATABASE_URL = postgres://yizthhwjzzmgrm:8d830f4e30e56aedc270b92fe7fdf98d083113b2169c99394e52d4a17ba4a808@ec2-50-19-222-129.compute-1.amazonaws.com:5432/d9bltlsuv2jqs
```
Comment (jema28) [inline config.env]: You shouldn't have your config.env up on github, please add this to your .gitignore.

File: package.json [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+  "name": "Artists8",
+  "version": "1.0.0",
+  "description": "## Description:",
+  "dependencies": {
+    "pg": "^7.12.0",
+    "env2": "^2.2.2"
+  },
+  "devDependencies": {},
+  "scripts": {
+    "test": "echo \"Error: no test specified\" && exit 1",
```
Comment (jema28) [inline package.json]: Please delete this as not being used.

File: src/app.js [hunk 3]
```diff
@@ -16,10 +17,20 @@ app.engine(
     extname: "hbs",
     layoutsDir: path.join(__dirname, "views", "layouts"),
     defaultLayout: "main",
-    partialsDir: path.join(__dirname, "views", "partials")
+    // partialsDir: path.join(__dirname, "views", "partials")
```
Comment (jema28) [inline src/app.js]: Why did you comment this out?

File: src/app.js [hunk 4]
```diff
@@ -16,10 +17,20 @@ app.engine(
...
   })
 );
-app.use(router)
 
 
+app.get("/", (req, res,next) => {
```
Comment (jema28) [inline src/app.js]: This method should be in a controller. Then write the method in the router index.js controller folders.

File: src/app.js [hunk 5]
```diff
@@ -16,10 +17,20 @@ app.engine(
...
+  get4painting((err,data)=>{
+    if(err)
+      next(err);
```
Comment (jema28) [inline src/app.js]: Where are you handling this? This will break if you don't have a global error middleware.

File: src/app.js [hunk 6]
```diff
@@ -16,10 +17,20 @@ app.engine(
...
+      console.log(data,"00000000")
```
Comment (jema28) [inline src/app.js]: Please delete this console.log.

File: src/database/db_build.sql [hunk 7]
```diff
@@ -0,0 +1,56 @@
+BEGIN;
```
Comment (jema28) [inline src/database/db_build.sql:1]: Please check this file in prettier.

File: src/database/db_build.sql [hunk 8]
```diff
@@ -0,0 +1,56 @@
...
+
+DROP TABLE IF EXISTS paintaings CASCADE;
+DROP TABLE IF EXISTS artists CASCADE;
+
+
+CREATE TABLE artists (
+    id SERIAL PRIMARY KEY,
+    name VARCHAR(150) NOT NULL,
```
Comment (jema28) [inline src/database/db_build.sql]: Should this be unique?

File: src/database/db_build.sql [hunk 9]
```diff
@@ -0,0 +1,56 @@
...
+    coutry VARCHAR(40)
```
Comment (jema28) [inline src/database/db_build.sql]: Spelling mistake: 'country'.

File: src/app.js [hunk 10]
```diff
@@ -16,10 +17,20 @@ app.engine(
...
     defaultLayout: "main",
-    partialsDir: path.join(__dirname, "views", "partials")
+    // partialsDir: path.join(__dirname, "views", "partials")
   })
 );
-app.use(router)
 
 
+app.get("/", (req, res,next) => {
+  get4painting((err,data)=>{
```
Comment (jema28) [inline src/app.js]: This should be called getExpensivePaintings.

File: src/database/db_build.sql [hunk 11]
```diff
@@ -0,0 +1,56 @@
...
+('5','johannes vermeer', 'Span'),
+('6','rembrandt', 'Netherlands'),
+('7','picasso', 'Netherlands'),
+('8','Edvard Munch', 'Norway'),
+('9','van gogh ', 'Netherlands')
+
+
+;
+
+CREATE TABLE paintaings (
```
Comment (jema28) [inline src/database/db_build.sql]: Spelling error: 'paintings'.

File: src/database/db_build.sql [hunk 12]
```diff
@@ -0,0 +1,56 @@
...
+    price double precision NOT NULL,
+    discription text ,
+    img text,
+    id_artist INTEGER,
+    FOREIGN KEY
+    (id_artist) REFERENCES artists (id)
+
+);
+
+INSERT INTO paintaings (name, price, discription,img, id_artist ) VALUES
```
Comment (jema28) [inline src/database/db_build.sql]: Same error.

File: src/database/db_build.sql [hunk 13]
```diff
@@ -0,0 +1,56 @@
...
+('9','van gogh ', 'Netherlands')
+
+
+;
+
+CREATE TABLE paintaings (
+    id SERIAL PRIMARY KEY,
+    name VARCHAR(150) NOT NULL,
+    price double precision NOT NULL,
+    discription text ,
```
Comment (jema28) [inline src/database/db_build.sql]: Spelling error: 'description'.

File: src/database/db_build.sql (hunk 12 above)
Comment (jema28) [inline src/database/db_build.sql]: Same error here also.

**Review by jema28 — CHANGES_REQUESTED** [review]

File: src/database/db_build.sql (hunk 12 above)
Comment (jema28) [inline src/database/db_build.sql]: Please look at the spacing here: your commas aren't in the right place.

File: src/queries/getData.js [hunk 14]
```diff
@@ -0,0 +1,15 @@
+const dbConnection = require("../database/db_connection.js");
+const get4painting = cb => {
+  dbConnection.query(
+    `select id,name,price,img from paintaings order By price desc limit 4;`,
```
Comment (jema28) [inline src/queries/getData.js]: Keep consistent here e.g. if you're using lowercase please write 'by'. You should also add spaces between the inputs e.g id, name, price

File: src/views/home.hbs [hunk 15]
```diff
@@ -0,0 +1,15 @@
+<input type="search">
+<h2>the most expensive 4 paintings</h2>
+<div>
+
```
Comment (jema28) [inline src/views/home.hbs]: Please delete this space.

File: src/views/home.hbs [hunk 16]
```diff
@@ -0,0 +1,15 @@
+<input type="search">
+<h2>the most expensive 4 paintings</h2>
```
Comment (jema28) [inline src/views/home.hbs]: 'The four most expensive paintings'

File: src/views/layouts/main.hbs [hunk 17]
```diff
@@ -0,0 +1,14 @@
+<!DOCTYPE html>
+<html>
+ <head>
+  <title>Country statistics</title>
+  <meta name="viewport" content="width=device-width, initial-scale=1">
+  <meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
+  <link href="//fonts.googleapis.com/css?family=Roboto:300,300i,400,400i,700,700i" rel="stylesheet">
+  <link href="styles.css" rel="stylesheet">
+  <!-- //web font -->
```
Comment (jema28) [inline src/views/layouts/main.hbs]: Please delete.

File: src/views/layouts/main.hbs [hunk 18]
```diff
@@ -0,0 +1,14 @@
+<!DOCTYPE html>
```
Comment (jema28) [inline src/views/layouts/main.hbs:1]: Is this eslinted?

File: src/views/home.hbs [hunk 19]
```diff
@@ -0,0 +1,15 @@
...
+<div>
+
+    {{#each paintings}}
+    <a href="/painting/:{{id}}">
+        <div>
```
Comment (jema28) [inline src/views/home.hbs]: You could change this to an article tag.

**Review by jema28 — CHANGES_REQUESTED** [review]

File: src/database/db_build.sql [hunk 20]
```diff
@@ -0,0 +1,48 @@
...
+DROP TABLE IF EXISTS paintings CASCADE;
+DROP TABLE IF EXISTS artists CASCADE;
+
+CREATE TABLE artists (
+    id SERIAL PRIMARY KEY, 
+    name VARCHAR(150) NOT NULL, 
+    coutry VARCHAR(40)
+);
+
+INSERT INTO artists (id, name, coutry) VALUES
```
Comment (95Rawan) [inline src/database/db_build.sql]: Why are you adding the id manually?

File: src/database/db_build.sql [hunk 21]
```diff
@@ -0,0 +1,48 @@
...
+    name VARCHAR(150) NOT NULL, 
+    price double precision NOT NULL, 
+    description text , 
+    img text, 
+    id_artist INTEGER, 
+    FOREIGN KEY
+    (id_artist) REFERENCES artists (id)
+);
+
+INSERT INTO paintings (name, price, description, img, id_artist) VALUES
```
Comment (95Rawan) [inline src/database/db_build.sql:35]: the id is manually added here too

File: config.env (hunk 1 above)
Comment (jema28) [reply config.env]: Please remove this file from the cache.

**Review by jema28 — CHANGES_REQUESTED** [review]

File: package.json [hunk 22]
```diff
@@ -0,0 +1,26 @@
+{
+  "name": "Artists8",
+  "version": "1.0.0",
+  "description": "## Description:",
+  "dependencies": {
```
Comment (jema28) [inline package.json:5]: Where are express and express-handlebars? These need to be added.

File: src/controllers/home.js [hunk 23]
```diff
@@ -0,0 +1,13 @@
+const get4ExpensivePainting = require('../../src/queries/getData');
+
+module.exports.get=("/", (req, res,next) => {
```
Comment (jema28) [inline src/controllers/home.js]: This should be `exports.get`.

File: src/controllers/index.js [hunk 24]
```diff
@@ -0,0 +1,9 @@
+const express = require("express")
+const router = express.Router();
+const home = require("./home")
+const path = require('path');
+router.use(express.json());
```
Comment (jema28) [inline src/controllers/index.js:9]: Please add space here.

File: src/database/db_build.js [hunk 25]
```diff
@@ -1,6 +1,6 @@
-const fs = require('fs');
+const fs = require("fs");
```
Comment (jema28) [inline src/database/db_build.js:1]: Please sort out your eslint!!!

**Review by jema28 — CHANGES_REQUESTED** [review]

File: src/app.js [hunk 26]
```diff
@@ -1,5 +1,6 @@
 const express = require("express");
 const path = require('path');
+const router = require('./controllers/index');
```
Comment (jema28) [inline src/app.js]: You don't need index here.

File: src/app.js [hunk 27]
```diff
@@ -1,5 +1,6 @@
...
 const exphbas = require("express-handlebars");
```
Comment (jema28) [inline src/app.js]: Spelling error here: 'exphbs'. Please change this in line 15 also.

File: package.json (hunk 22 above)
Comment (jema28) [reply package.json:5]: I think this is a problem with your version of node/npm, you may have to update them.

**Review by jema28 — CHANGES_REQUESTED** [review]

**Review by jema28 — APPROVED** [review]

**Pull request merged**